cli/src/command.rs

```diff
@@ -154,6 +155,39 @@ pub fn run() -> Result<()> {
 			let runner = cli.create_runner(cmd)?;
 			runner.sync_run(|config| cmd.run(config.chain_spec, config.network))
 		},
+		Some(Subcommand::BuildSyncSpec(cmd)) => {
+			let runner = cli.create_runner(cmd)?;
+			let chain_spec = &runner.config().chain_spec;
+
+			set_default_ss58_version(chain_spec);
+
+			let authority_discovery_enabled = cli.run.authority_discovery_enabled;
+			let grandpa_pause = if cli.run.grandpa_pause.is_empty() {
+				None
+			} else {
+				Some((cli.run.grandpa_pause[0], cli.run.grandpa_pause[1]))
+			};
+
+			if chain_spec.is_kusama() {
+				info!("----------------------------");
+				info!("This chain is not in any way");
+				info!("      endorsed by the       ");
+				info!("     KUSAMA FOUNDATION      ");
+				info!("----------------------------");
+			}
+
+			runner.async_run(|config| {
+				let chain_spec = config.chain_spec.cloned_box();
+				let network_config = config.network.clone();
+				let service::NewFull { task_manager, client, network_status_sinks, .. }
+					= service::new_full_nongeneric(
+						config, None, authority_discovery_enabled, grandpa_pause, false,
+					)?;
+				let client = Arc::new(client);
+
+				Ok((cmd.run(chain_spec, network_config, client, network_status_sinks), task_manager))
```

Comment:
why not just pass the `Configuration` struct? `chain_spec` and `network_config` live on the `Configuration` object

Reply:
The `Configuration` gets consumed by `new_full_nongeneric`, so it can't be used.

Reply:
I would argue that it should be clone, but that's for another time. This looks solid to me